src/main/java/hudson/remoting/Launcher.java

```diff
@@ -379,17 +380,23 @@ private void runAsTcpServer() throws IOException, InterruptedException {
 
         // write a port file to report the port number
         FileWriter w = new FileWriter(tcpPortFile);
-        w.write(String.valueOf(ss.getLocalPort()));
-        w.close();
-
+        try {
+            w.write(String.valueOf(ss.getLocalPort()));
+        } finally {
+            w.close();
+        }
+        
         // accept just one connection and that's it.
         // when we are done, remove the port file to avoid stale port file
         Socket s;
         try {
             s = ss.accept();
             ss.close();
         } finally {
-            tcpPortFile.delete();
+            boolean deleted = tcpPortFile.delete();
```

Comment:
BTW - java 7 NIO is much better as it will tell you why it could not delete the file...